src/utils/components/DiskModal/utils/form.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  getBootDisk,
+  getDataVolumeTemplates,
+  getDisks,
+  getVolumes,
+} from '@kubevirt-utils/resources/vm';
+import { isInstanceTypeVM } from '@kubevirt-utils/resources/vm/utils/instanceTypes';
+import { generatePrettyName, isEmpty } from '@kubevirt-utils/utils/utils';
+import { isRunning } from '@virtualmachines/utils';
+
+import { DEFAULT_DISK_SIZE } from './constants';
+import { requiresDataVolume } from './helpers';
+import { InterfaceTypes, SourceTypes, V1DiskFormState } from './types';
+
+const getDefaultDataVolumeTemplate = (name: string): V1DataVolumeTemplateSpec => ({
+  metadata: { name },
+  spec: { source: {}, storage: { resources: { requests: { storage: DEFAULT_DISK_SIZE } } } },
+});
+
+const createInitialStateFromSource: Record<
+  Exclude<SourceTypes, SourceTypes.OTHER>,
+  (volume: V1Volume, dataVolumeTemplate: V1DataVolumeTemplateSpec) => void
+> = {
+  [SourceTypes.BLANK]: (volume: V1Volume, dataVolumeTemplate: V1DataVolumeTemplateSpec) =>
+    (dataVolumeTemplate.spec.source.blank = {}),
+  [SourceTypes.CLONE_PVC]: (volume: V1Volume, dataVolumeTemplate: V1DataVolumeTemplateSpec) =>
+    (dataVolumeTemplate.spec.source.pvc = { name: '', namespace: '' }),
+  [SourceTypes.DATA_SOURCE]: (volume: V1Volume, dataVolumeTemplate: V1DataVolumeTemplateSpec) =>
+    (dataVolumeTemplate.spec.sourceRef = { kind: DataSourceModel.kind, name: '', namespace: '' }),
+  [SourceTypes.EPHEMERAL]: (volume: V1Volume) => (volume.containerDisk = { image: '' }),
+  [SourceTypes.HTTP]: (volume: V1Volume, dataVolumeTemplate: V1DataVolumeTemplateSpec) =>
+    (dataVolumeTemplate.spec.source.http = { url: '' }),
+  [SourceTypes.PVC]: (volume: V1Volume) => (volume.persistentVolumeClaim = { claimName: '' }),
+  [SourceTypes.REGISTRY]: (volume: V1Volume, dataVolumeTemplate: V1DataVolumeTemplateSpec) =>
+    (dataVolumeTemplate.spec.source.registry = { url: '' }),
+  [SourceTypes.UPLOAD]: (volume: V1Volume, dataVolumeTemplate: V1DataVolumeTemplateSpec) =>
+    (dataVolumeTemplate.spec.source.upload = {}),
+};
+
+export const getDefaultEditValues = (vm: V1VirtualMachine, editDiskName?: string) => {
+  const isBootSource = getBootDisk(vm)?.name === editDiskName;
+  let disk = getDisks(vm)?.find((d) => d.name === editDiskName);
+  const volume = getVolumes(vm)?.find((v) => v.name === editDiskName);
+  const dataVolumeTemplate = getDataVolumeTemplates(vm)?.find(
+    (dv) => getName(dv) === volume?.dataVolume?.name,
+  );
+
+  if (isEmpty(disk) && isInstanceTypeVM(vm)) disk = { name: editDiskName };
+
+  return {
+    dataVolumeTemplate,
+    disk,
+    isBootSource,
+    volume,
+  };
+};
+
+export const getDefaultCreateValues = (
+  vm: V1VirtualMachine,
+  createDiskSource: SourceTypes,
+): V1DiskFormState => {
+  const newDiskName = generatePrettyName('disk');
+  const newDataVolumeName = generatePrettyName(`dv-${getName(vm)}`);
+
+  const withDataVolume = requiresDataVolume(createDiskSource);
```

Comment:
requiresDataVolume its hard  to unserstand what the function does from the naming, please consider changing the naming